python/py-numpy/Portfile

```diff
@@ -286,38 +72,89 @@ variant."
             reinplace   "s|---|\\\\.f|" ${worksrcpath}/f-wrapper
             reinplace   "s|&&&|${prefix}|" ${worksrcpath}/f-wrapper
         }
+    }
 
-        if {[variant_isset atlas]} {
-            # We must link against libSatlas or libTatlas, not libAtlas
-            if {[file exists ${prefix}/lib/libtatlas.dylib]} {
-                reinplace -E \
-                    "s|_lib_atlas = \\\['atlas'\\\]|_lib_atlas = \\\['tatlas'\\\]|" \
-                    ${worksrcpath}/numpy/distutils/system_info.py
-            } elseif {[file exists ${prefix}/lib/libsatlas.dylib]} {
-                reinplace -E \
-                    "s|_lib_atlas = \\\['atlas'\\\]|_lib_atlas = \\\['satlas'\\\]|" \
-                    ${worksrcpath}/numpy/distutils/system_info.py
-            } else {
-                return -code error "Unable to find Atlas dylibs. Bailing out."
-            }
+    depends_build-append \
+                        path:bin/pkg-config:pkgconfig
+
+    build.env-append    CYTHON=${prefix}/bin/cython-${python.branch}
+
+    # do not allow fallback to (slow) NumPy-internal routines if no external BLAS/LAPACK libraries are found
+    build.args-append -Csetup-args=-Dallow-noblas=false
+
+    compilers.setup     -clang -gcc44 -gcc45 \
+                        -gcc46 -gcc47 -gcc48 -g95
+
+    # https://trac.macports.org/ticket/67136
+    depends_run-append  port:py${python.version}-oldest-supported-numpy
+
+    depends_lib-append  port:fftw-3 \
+                        path:bin/cython-${python.branch}:py${python.version}-cython
+
+    variant accelerate conflicts openblas flexiblas blis description "Use Apples Accelerate Framework" {
+        build.args-append -Csetup-args=-Dblas=accelerate -Csetup-args=-Dlapack=accelerate
+    }
+
+    variant openblas conflicts accelerate flexiblas blis description "Use MacPorts OpenBLAS Libraries" {
+        depends_lib-append  path:lib/libopenblas.dylib:OpenBLAS
+
+        # see: https://trac.macports.org/ticket/66685
+        supported_archs     arm64 ppc ppc64 x86_64
+
+        build.args-append -Csetup-args=-Dblas=openblas -Csetup-args=-Dlapack=openblas \
+                          -Csetup-args=-Dpkg_config_path=${prefix}/lib/pkgconfig
+    }
+
+    variant flexiblas conflicts accelerate openblas blis description "Use MacPorts FlexiBLAS libraries" {
+        depends_lib-append  port:flexiblas
+
+        build.args-append -Csetup-args=-Dblas=flexiblas -Csetup-args=-Dlapack=flexiblas \
+                          -Csetup-args=-Dpkg_config_path=${prefix}/lib/pkgconfig
+    }
+
+    variant blis conflicts accelerate openblas flexiblas description "Use MacPorts BLIS libraries" {
```

Comment:
Does it use lapack? Blis does not have its own lapack (unlike OpenBLAS, for example), so for lapack it needs either its native (but partial) implementation in `libflame` or standalone `lapack` (there is a port for it).

Reply:
Good question ; I don't know. Let me try this tonight without anything else installed and see how the tests go.

Reply:
Yeah, it does need a LAPACK library. I set it naively to `blis` and thus `meson` did not complain but it effectively used  Apple's Accelerate framework. So not what I intended.... NumPy claims that it can use `libflame` in its release notes (since v. 1.17 or so) but then doesn't list it as an option in the `meson_options.txt`. When I try to just give it as an option, `meson` cannot find it as there is no `.pc` or `.cmake` file installed for `libflame`. 

I looked briefly at `libflame` and tried to update it to the latest version (and build a shared library), but both the latest commit hash and the current MacPorts version don't build for me from source. Regardless, I am also not sure whether there is a `.pc` file included in the most recent commit hash.

It seems like some/most of these issue have been resolved in NumPy v2 so I am not sure whether it's worth spending too much time on this right now. If we keep the `blis` variant then I would prefer to have `libflame` used as the LAPACK implementation but for that the first step would be to manually craft a `.pc` file and see if NumPy will use it. Not sure if @barracuda156 would be willing to take a closer look at `libflame`?